packages/mobile/src/navigator/NavigationService.ts

```diff
@@ -89,12 +89,24 @@ export function navigate<RouteName extends keyof StackParamList>(
     .then(() => {
       Logger.debug(`${TAG}@navigate`, `Dispatch ${routeName}`)
 
-      navigationRef.current?.dispatch(
-        NavigationActions.navigate({
-          routeName,
-          params,
-        })
-      )
+      // Remove the WC loading screen from the stack as we never want to land
+      // on this screen when navigating back. Ensures that the dapp webview WC
+      // navigation works correctly.
+      if (navigationRef.current?.getCurrentRoute()?.name === Screens.WalletConnectLoading) {
```

Comment:
Is there a way to make this logic more local?
This works but feels a bit hacky to be applied here.

But I feel the alternative is probably not great either 😄